_episodes_rmd/06_day5.Rmd

```diff
@@ -0,0 +1,266 @@
+> WIcities_points <- st_as_sf(WIcities, coords=c('LONGITUDE','LATITUDE'), crs=crs(maxTemp_global))
+> 
+> # check class of object - should be "sf" data frame
+> class(WIcities_points)
+> ```
+{: .solution}
+
+A critical part of working with raster and vector data together is that the datasets are in the same coordinate system. 
+
+Does it make sense? Are they the same? You can use the functions in the `raster` package to check out your data.
+
+```{r check-coordinate-systems}
+
+crs(maxTemp_global)
+crs(WIcities_points)
+
+extent(maxTemp_global)
+extent(WIcities_points)
+
+res(maxTemp_global)
+
+```
+
+
+Finally, let's subset our spatial data to only include the city points we are interested in. For the sample dataset, we are going to subset the data to only include the cities of Madison and Milwaukee.
+
+If you are using your own data, you may or may not need to do this subset.
+
+> ## Solution
+> ```{r subset-vector-data}
+> 
+> WIcities_points$NAME
+> 
+> MM_points <- WIcities_points[WIcities_points$NAME=='Madison'|WIcities_points$NAME=='Milwaukee',]
+> ```
+{: .solution}
+
+# 3) Crop raster with vector 
+
+It may be useful to limit the raster data to only areas of interest. This is helpful when you load a larger raster file, such as data for the entire globe, but are only interested in a small part. For our capstone data, we loaded the global temperature data, but we are only interested in the data for Wisconsin, and specifically, the data for Milwaukee and Madison. Let's use the `crop()` function in the `raster` package to cut out the maximum temperature data to only the area for the Wisconsin cities.
+
+> ## Solution
+> ```{r crop-raster-by-vector}
+> 
+> maxTemp_WI <- crop(x = maxTemp_global, y = extent(WIcities_points),snap="out")
+> maxTemp_WI_df <- as.data.frame(maxTemp_WI, xy=T)
+> 
+> ggplot() +
+>   geom_raster(data = maxTemp_WI_df,aes(x = x, y = y, fill = bio5)) + 
+>   geom_sf(data = MM_points,color="yellow",size=5) + 
+>   coord_sf()
+> ```
+{: .solution}
+
+# 4) Summarize raster data by vector locations
+Now that we have our raster layer that contains the data we want to summarize, and the vector data that contains the locations indicating where we want to summarize the data, we can now put them together.
+
+If you are working with your own data, ask yourself:
+
++ How do I want to summarize the data? For example, do I want a single mean value? A maximum? A distribution of all values?
++ What is the area I want to summarize? If it's a point, do you want a certain distance around the point? Or, if it's an area, you will include all of the raster values within that area.
+
+## Extract pixel values
+
+For the capstone data, we want to extract all the pixel values within a distance buffer of the two city points. Since our data is "unprojected" and therefor the units are in lat/long, the buffer distance will be in meters (hint: this information is in the Help file for the `extract` function!)
```

Comment:
```suggestion
For the capstone data, we want to extract all the pixel values within a distance buffer of the two city points. Since our data is "unprojected" and therefore the units are in latitude and longitude, the buffer distance will be in meters (hint: this information is in the Help file for the `extract` function!)
```